dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/model/Contentlet.java

```diff
@@ -124,15 +125,21 @@ public String getName() {
 
     @Override
     public String getTitle(){
-    	if(map.get("title") !=null){
-    		return map.get("title").toString();
-    	}
-
     	try {
-    	    ContentletAPI conAPI = APILocator.getContentletAPI();
-    	    String x = conAPI.getName(this, APILocator.getUserAPI().getSystemUser(), false);
-    	    map.put("title", x);
-    	    return x;
+
+    		//Verifies if the content type has defined a title field
+			List<String> fields = this.getContentType().fields().stream().
+					map(com.dotcms.contenttype.model.field.Field::variable).filter(variable -> variable.equals("title")).collect(Collectors.toList());
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Specify a type for: 'variable' [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2211)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move the "title" string literal on the left side of this string comparison. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1132)

Reply:
You might want to change the collect by findFirst, it will be more efficient since the loop will be break when it finds the first, in the worst case will be the same loop.
And instead of asking for null or empty, you can ask if the optional is present